crates/proto/src/op/message.rs

```diff
@@ -914,6 +915,19 @@ fn test_emit_and_read_records() {
     test_emit_and_read(message);
 }
 
+#[test]
+fn test_extended_response_code() {
+    let mut message = Message::new();
+    message.set_response_code(ResponseCode::BADCOOKIE);
+    test_emit_and_read(message);
```

Comment:
The tests currently fail, since the reparsed message does not contain an additional entry whereas the original message does contain one. This is the OPT record for EDNS.

Some feedback why parsing a Message removes the OPT record but correctly extracts the EDNS data would be appreciated.

Reply:
That sounds a little confusing, I'm not sure why that would be different. Is it possible there is an ordering issue on calculating the counts? One possibility is that some of the processing logic is dropping EDNS due to mishandling this particular ResponseCode or something?

Reply:
I looked at the PR again and now found that I probably need to call `update_counts` before `test_emit_and_read`.u